sentry-types/src/protocol/attachment.rs

```diff
@@ -0,0 +1,80 @@
+use std::fmt;
+
+/// The different types an attachment can have
+#[derive(Debug, Copy, Clone, PartialEq)]
+pub enum AttachmentType {
+    /// A standard attachment without special meaning. (default)
+    Attachment,
+    /// A minidump file that creates an error event and is symbolicated. The
+    /// file should start with the `MDMP` magic bytes.
+    Minidump,
+    /// An Apple crash report file that creates an error event and is symbolicated.
+    AppleCrashReport,
+    /// An XML file containing UE4 crash meta data. During event ingestion,
+    /// event contexts and extra fields are extracted from this file.
+    UnrealContext,
+    /// A plain-text log file obtained from UE4 crashes. During event ingestion,
+    /// the last logs are extracted into event breadcrumbs.
+    UnrealLogs,
+}
+
+impl Default for AttachmentType {
+    fn default() -> Self {
+        Self::Attachment
+    }
+}
+
+impl AttachmentType {
+    /// Gets the string value Sentry expects for the attachment type
+    pub fn as_str(self) -> &'static str {
+        match self {
+            Self::Attachment => "event.attachment",
+            Self::Minidump => "event.minidump",
+            Self::AppleCrashReport => "event.applecrashreport",
+            Self::UnrealContext => "unreal.context",
+            Self::UnrealLogs => "unreal.logs",
+        }
+    }
+}
+
+#[derive(PartialEq)]
+/// Represents an attachment item
+pub struct Attachment {
+    /// The actual attachment data
+    pub buffer: std::borrow::Cow<'static, [u8]>,
+    /// The filename of the attachment
+    pub filename: std::ffi::OsString,
```

Comment:
```suggestion
    pub filename: String,
```

Since this is about the envelope header which will be sent as JSON, this should be independent of FS encoding.